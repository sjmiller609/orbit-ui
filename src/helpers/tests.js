import { BrowserRouter } from 'react-router-dom'
import { shape } from 'prop-types'
import { mount } from 'enzyme'

// Instantiate router context
const router = {
  history: new BrowserRouter().history,
  route: {
    location: {},
    match: {},
  },
}

const createContext = () => ({
  context: { router },
  childContextTypes: { router: shape({}) },
})

// wrap with router context - to mount RR links correctly
export function mountWrap(node) {
  return mount(node, createContext())
}
