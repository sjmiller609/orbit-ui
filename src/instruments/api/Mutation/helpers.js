const has = (str, find) => ~str.toLowerCase().indexOf(find.toLowerCase())

export const formErrors = ({ error, errors }) => {
  if (!error) return

  let err = JSON.stringify(error).toLowerCase()
  return errors.find(e => {
    if (e.key && has(err, e.key)) return e
    else if (e.allKeys && e.allKeys.every(k => has(err, k))) return e
    else if (e.someKeys && e.someKeys.some(k => has(err, k))) return e
  })
}
