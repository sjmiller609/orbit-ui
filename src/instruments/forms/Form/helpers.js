// faltten and stringify nested keys
export const unpack = (obj, ns, hash) => {
  if (!obj) return
  return Object.entries(obj).reduce((acc, [key, value]) => {
    const keypath = ns ? `${ns}.${key}` : key
    if (value && typeof value === 'object') {
      return unpack(value, keypath, acc)
    }
    acc[keypath] = value
    return acc
  }, hash || {})
}

const roll = (keys, obj, value) => {
  const key = keys[0]
  // case 1 - not obj
  if (keys.length === 1) {
    obj[key] = value
    return obj
  }
  // case 2 - doesn't exist
  if (!obj[key]) {
    obj[key] = isNaN(keys[1]) ? {} : []
  }
  // case 3 - exists
  obj[key] = roll(
    keys.slice(1),
    Array.isArray(obj[key]) ? [...obj[key]] : { ...obj[key] },
    value
  )

  return obj
}

// convert back into obj
export const pack = (obj, type = {}) => {
  return Object.entries(obj).reduce((acc, [key, value]) => {
    const keys = key.split('.')

    return roll(keys, acc, value)
  }, type)
}

export const packChild = ({ name, obj }) => {
  const obj2 = {}
  let isArray
  Object.entries(obj).forEach(k => {
    if (name.length < k[0].length && k[0].indexOf(name) === 0) {
      const n = k[0].slice(name.length + 1)
      // must be true for all matching
      isArray = !isNaN(n.slice(0, n.indexOf('.')))
      obj2[n] = k[1]
    }
  })
  if (Object.keys(obj2).length) return pack(obj2, isArray ? [] : {})
  return null
}

export const removeChild = ({ name, obj = {} }) => {
  const obj2 = {}
  Object.entries(obj).forEach(k => {
    if (name.length >= k[0].length || k[0].indexOf(name) !== 0) {
      obj2[k[0]] = k[1]
    }
  })
  return obj2
}
