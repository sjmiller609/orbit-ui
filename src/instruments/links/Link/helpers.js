// check if url is external
export const externalUrl = url => {
  if (typeof url === 'object') return false
  const path = /^https?:\/\/|^\/\//i
  if (path.test(url) && !~url.indexOf(window.location.origin)) return true
  return
}
