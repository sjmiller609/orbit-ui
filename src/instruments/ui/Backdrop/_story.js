import React from 'react'
import { storiesOf } from '@storybook/react'
import { Backdrop } from 'instruments'

storiesOf('Instruments|UI', module).add('Backdrop', () => (
  <Backdrop show={true} />
))
