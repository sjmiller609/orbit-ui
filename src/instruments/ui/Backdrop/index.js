'use strict'
import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'

import s from './styles.scss'

// Full screen background for modals, dropdowns, etc
const Backdrop = ({ show, close, blur }) => {
  if (!show) return null
  return (
    <div
      className={classnames(
        s.backdrop,
        blur ? s.blur : null,
        close ? s.close : null
      )}>
      <div onClick={close} />
    </div>
  )
}

Backdrop.propTypes = {
  show: PropTypes.bool,
  blur: PropTypes.bool,
  close: PropTypes.func,
}

export default Backdrop
