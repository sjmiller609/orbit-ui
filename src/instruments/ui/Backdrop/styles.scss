@import 'src/styles/mixins.scss';

.backdrop {
  position: relative;
  z-index: inherit;

  div {
    position: fixed;
    width: 100vw;
    height: 100vh;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    z-index: inherit;
  }
  &.blur div {
    background: $purple;
    opacity: 0.5;
  }
  &.close {
    cursor: pointer;
  }
  // @media screen and (max-width:$mobileNav) {
  //   div {
  //     background: $darkWhite;
  //     opacity: .7;
  //   }
  // }
}
