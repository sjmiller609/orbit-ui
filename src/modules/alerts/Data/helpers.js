import { formErrors } from 'instruments'

const errors = [
  {
    key: 'permissions',
    name: 'properties.alert_emails',
    error: 'Contact your workspace admin to upgrade your permissions.',
  },
]

export const handleError = error => formErrors({ error, errors })

export const trimError = error => error.split(':')[1].trim()
