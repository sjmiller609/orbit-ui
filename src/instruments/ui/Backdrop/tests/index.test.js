import React from 'react'
import { mountWrap } from 'helpers/tests'
import Backdrop from '../index'

const props = {
  show: false,
  blur: false,
  close: jest.fn(),
}

describe('Backdrop', () => {
  it('should be defined', () => {
    expect(Backdrop).toBeDefined()
  })

  it('renders correctly', () => {
    const wrapper = mountWrap(<Backdrop {...props} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should open when show is true', () => {
    const wrapper = mountWrap(<Backdrop {...props} show={true} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should handle blur', () => {
    const wrapper = mountWrap(<Backdrop {...props} show={true} blur={true} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should handle hide close', () => {
    const wrapper = mountWrap(<Backdrop {...props} show={true} close={null} />)
    expect(wrapper).toMatchSnapshot()
  })
})
