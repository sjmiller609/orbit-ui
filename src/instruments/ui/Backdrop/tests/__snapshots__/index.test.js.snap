// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Backdrop renders correctly 1`] = `
<Backdrop
  blur={false}
  close={[Function]}
  show={false}
/>
`;

exports[`Backdrop should handle blur 1`] = `
<Backdrop
  blur={true}
  close={[Function]}
  show={true}
>
  <div
    className="backdrop blur close"
  >
    <div
      onClick={[Function]}
    />
  </div>
</Backdrop>
`;

exports[`Backdrop should handle hide close 1`] = `
<Backdrop
  blur={false}
  close={null}
  show={true}
>
  <div
    className="backdrop"
  >
    <div
      onClick={null}
    />
  </div>
</Backdrop>
`;

exports[`Backdrop should open when show is true 1`] = `
<Backdrop
  blur={false}
  close={[Function]}
  show={true}
>
  <div
    className="backdrop close"
  >
    <div
      onClick={[Function]}
    />
  </div>
</Backdrop>
`;
